2주차 과제1

moon-kun86 opened the pull request:
1. 2주차 과제1 제출합니다. 
2.export default 안넣으면 안되는걸 한참 찾아봤습니다. 리뷰 부탁드리겠습니다.

moon-kun86 committed:
2주차 과제1

heyjyu commented:
그러셨군요! 뭔가 안될 때 해결하는 방법을 터득하신 것 같네요! export default 와 export의 차이를 혹시 아직 모르신다면 이 부분도 함께 공부해보시면 도움이 될 것 같아요!

heyjyu commented:
vs code를 사용하신다면 eslint 익스텐션을 설치해보시면 좋을 것 같습니다! 파일의 마지막 코드 뒤에는 한 줄을 띄워서 빈 줄을 하나 넣어주시는 게 좋습니다!

heyjyu commented on src/component/App.jsx:
```diff
@@ -0,0 +1,29 @@
+/* @jsx React.createElement */
+
+import React, { useState } from 'react'; // 리액트 임포트 필수! useState를 사용하므로 필수로 넣어줘야함
```
리액트 17버전부터는 React를 import하지 않아도 됩니다! 한 번 어떻게 하면 되는지 찾아보시는 것도 좋은 공부가 되실 것 같습니다!

moon-kun86 replied:
네 한번 찾아보겠습니다.

heyjyu commented on src/component/App.jsx:
```diff
@@ -0,0 +1,29 @@
+function App() {
+    const [state, setState]  = useState({
+        count: 0,
+    });
```
state에 여러 값이 들어올 예정일까요~? 만약 아니라면 [count, setCount]로 받아서 코드를 작성하시면 작성하신 의도를 파악하기 더 쉬운 가독성 좋은 코드가 될 것 같습니다!

moon-kun86 replied:
아니요 현재는 하나의 값만 들어올 예정입니다. 아샬님의 강의를 보고 따라한겁니다.
알려주는 코드로 수정을 해보겠습니다.

heyjyu replied:
아샬님께서 보통 써주시는 코드가 정말 잘 짜여진 코드이긴 해서 그대로 가져와서 쓰고, 익히는 연습도 도움이 많이 되실 거예요! 그렇지만 추가적으로 개선할 수 있는 방안이 있을 수 있으니 스스로도 PR을 올린 후 셀프리뷰를 하면서 리팩토링하는 연습을 해보시는 것도 많은도움이 될 것 같아요!

moon-kun86 committed:
Update App.jsx

AddNumber를 추가여 버튼 클릭 시 마다 클릭미의 카운터가 더해지도록 추가하였습니다.

moon-kun86 committed:
Update Button.jsx

// eslint-disable-next-line react/prop-types eslint 오류 수정을 위해 추가
onClick 함수를 매개 변수로 넣어서 AddNumber를 App.jsx에서 사용할 수 있도로 추가하였습니다.

moon-kun86 committed:
Update Buttons.jsx

moon-kun86 committed:
Update Counter.jsx

moon-kun86 committed:
Update Page.jsx

moon-kun86 commented:
버튼 클릭 시 카운터에 추가 되도록 수정하였습니다.
eslint에서도 아무런 에러가 없었습니다. 
그러나 테스트를 했을때 모든 인수테스트가 실패로 나왔습니다. 찾아보고 있긴한데 왜그런지 궁금합니다.

heyjyu commented:
> 버튼 클릭 시 카운터에 추가 되도록 수정하였습니다. eslint에서도 아무런 에러가 없었습니다. 그러나 테스트를 했을때 모든 인수테스트가 실패로 나왔습니다. 찾아보고 있긴한데 왜그런지 궁금합니다.


코드를 잘 작성하셨고, 과제가 원하는 바도 잘 이끌어 내신 것 같아요!
테스트가 통과되지 않았던 것은 테스트는 (0) 이 있는지 찾고 있는데, 코드를 작성하실 때 디자인적인 측면을 고려하신 건지 ( 0 )으로 작성을 하셔서 통과가 안되고 있었던 것 뿐입니다!

테스트는 test디렉토리의 simple_test.js를 보시면 어떤식으로 테스트가 진행이되고 있고, 내 코드에서 어떤 게 안 맞았는지 테스트 코드를 보시고 확인해보실 수도 있습니다!

heyjyu commented on src/component/App.jsx:
```diff
@@ -0,0 +1,31 @@
+
+}
+
+export default App;
```
export default를 함수를 선언할 때 써주시면 더 간결한 코드가 되어 좋을 것 같습니다!

moon-kun86 commented:
> > 버튼 클릭 시 카운터에 추가 되도록 수정하였습니다. eslint에서도 아무런 에러가 없었습니다. 그러나 테스트를 했을때 모든 인수테스트가 실패로 나왔습니다. 찾아보고 있긴한데 왜그런지 궁금합니다.
> 
> 코드를 잘 작성하셨고, 과제가 원하는 바도 잘 이끌어 내신 것 같아요! 테스트가 통과되지 않았던 것은 테스트는 (0) 이 있는지 찾고 있는데, 코드를 작성하실 때 디자인적인 측면을 고려하신 건지 ( 0 )으로 작성을 하셔서 통과가 안되고 있었던 것 뿐입니다!
> 
> 테스트는 test디렉토리의 simple_test.js를 보시면 어떤식으로 테스트가 진행이되고 있고, 내 코드에서 어떤 게 안 맞았는지 테스트 코드를 보시고 확인해보실 수도 있습니다!

네 한번 확인해보겠습니다.

moon-kun86 committed:
Update App.jsx

export default App; ->> export default 함수 앞에 선언해주면서 가독성을 높여주었습니다.

moon-kun86 committed:
Update Button.jsx

export default Button; ->> export default 함수 앞에 선언해주면서 가독성을 높여주었습니다.

moon-kun86 committed:
Update Buttons.jsx

export default를 함수 앞에 선언하여 가독성을 높여주었습니다.

moon-kun86 committed:
Update Counter.jsx

export default를 함수 앞에 선언하여 가독성을 높여주었습니다.

moon-kun86 committed:
Update Page.jsx

export default를 함수 앞에 선언하여 가독성을 높여주었습니다.

moon-kun86 commented:
> > > 버튼 클릭 시 카운터에 추가 되도록 수정하였습니다. eslint에서도 아무런 에러가 없었습니다. 그러나 테스트를 했을때 모든 인수테스트가 실패로 나왔습니다. 찾아보고 있긴한데 왜그런지 궁금합니다.
> > 
> > 
> > 코드를 잘 작성하셨고, 과제가 원하는 바도 잘 이끌어 내신 것 같아요! 테스트가 통과되지 않았던 것은 테스트는 (0) 이 있는지 찾고 있는데, 코드를 작성하실 때 디자인적인 측면을 고려하신 건지 ( 0 )으로 작성을 하셔서 통과가 안되고 있었던 것 뿐입니다!
> > 테스트는 test디렉토리의 simple_test.js를 보시면 어떤식으로 테스트가 진행이되고 있고, 내 코드에서 어떤 게 안 맞았는지 테스트 코드를 보시고 확인해보실 수도 있습니다!
> 
> 네 한번 확인해보겠습니다.

>"테스트는 (0) 이 있는지 찾고 있는데, 코드를 작성하실 때 디자인적인 측면을 고려하신 건지 ( 0 )으로 작성" ->> 이부분이 이해가 가지 않아서 질문드립니다. 
--> count를 0이라고 선언을 해줬는데 테스트에서 요구하는(0)이 있는게 아닌가요?>

heyjyu commented:
> 

괄호와 숫자 사이에 띄어쓰기가 있고 없고의 차이입니다! 정확히 동일한지를 확인하기 때문에 띄어쓰기 하나만 달라도 다르게 인식하게 됩니다~

heyjyu commented on src/component/App.jsx:
```diff
@@ -0,0 +1,29 @@
+    const { count } = state;
+
+    function handleClick(AddNumber = 1) { // AddNumber는 기본값 1을 통하여 클릭미 클릭할때 1씩 증감하도록 처리
+        console.log(AddNumber); // onClick 함수에서 가져온 AddNumber = children이다 children은 button  컴포넌트에서 가져온거다.
```
로그를 찍는 부분은 사용하신 후에는 지워주시면 좋을 것 같아요!
